### PR TITLE
Modify callback for image find_dimension....

### PR DESCRIPTION
**Pull request by aishmita-aggarwal**
update specs for search/base and image

Note: Fix for Exception raised on image upload with invalid content type.

### CODE REVIEW

File: core/app/models/spree/image.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ class Image < Asset
 
     # save the w,h of the original image (from which others can be calculated)
     # we need to look at the write-queue for images which have not been saved yet
-    after_post_process :find_dimensions
+    before_save :find_dimensions
```
Comment (nishant-cyro) [inline core/app/models/spree/image.rb]: @aishmita-aggarwal @priyank-gupta - I guess we should not run this callback every time an Image object is updated, we should run this only when an attachment is updated in Image object.

**Comment by nishant-cyro** [comment]
@aishmita-aggarwal @priyank-gupta - :+1:

**Comment by priyank-gupta** [comment]
:+1:

**Comment by mbj** [comment]
The spec does not reproduce the failure case (invalid content type) and and just re-specifies active record DSL semantics. -1 from me. 

Add a spec that reproduces the problem with a semantic change that fixes the problem to get a +1 from me.

**Comment by hoshinotsuyoshi** [comment]
https://github.com/spree/spree/pull/6988 は 3.1.0.rc1では入っているので、 3.1.0との比較ということにします!

**Pull request merged**